### DISCUSSION TITLE
How to apply a ThemeResource to a Textblock created during runtime

### DISCUSSION THREAD

**Discussion by roshannzth** [microsoft/microsoft-ui-xaml#9016] [Q&A]
`<TextBlock Text="Sample Text" Foreground="{ThemeResource TextFillColorSecondary}"></TextBlock>`

I have a sample textblock xaml code as above.,
I want to create this control during runtime and adhere to themes on runtime.
What i tried here is 

```
Controls::TextBlock textBlock{}
auto foregroundBrush = Application::Current().Resources().Lookup(box_value(L"TextFillColorSecondary")).as<winrt::Windows::UI::Color>();
auto color           = Media::SolidColorBrush(foregroundBrush);
textblock.Foreground(color);
```

This behaves like a StaticResource which adapts to the theme once on startup. 
(If i start my application in dark mode, text is in white color, and i switch to light theme, text remains in white )

I tried to create a CustomStyle.xaml and included the following resource

```
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">
    <SolidColorBrush x:Key="TextFillColorPrimaryTheme" Color="{ThemeResource TextFillColorPrimary}" />
</ResourceDictionary>
```

and added this info in App.xaml
```
<Application.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <XamlControlsResources xmlns="using:Microsoft.UI.Xaml.Controls" />
                <!-- Other merged dictionaries here -->
                <ResourceDictionary Source="/CustomStyles.xaml" />
            </ResourceDictionary.MergedDictionaries>
            <!-- Other app resources here -->
        </ResourceDictionary>
    </Application.Resources>
```

But when i run the app, app crashes with the following error message
`L"Cannot find a Resource with the Name/Key TextFillColorPrimary`

Any idea what i am doing wrong, or is there a way to apply theme resource from code ?

**Comment by asklar** [comment]
See #4443 and other issues linked to it
